packages/mongodb-access-adapter/src/test/kotlin/com/mongodb/jbplugin/accessadapter/slice/BuildInfoTest.kt

```diff
@@ -1,22 +1,48 @@
 package com.mongodb.jbplugin.accessadapter.slice
 
+import com.mongodb.ConnectionString
+import com.mongodb.client.model.Filters
 import com.mongodb.jbplugin.accessadapter.MongoDbDriver
+import com.mongodb.jbplugin.accessadapter.toNs
 import org.bson.Document
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Test
 import org.mockito.Mockito
+import org.mockito.Mockito.`when`
 
 import kotlinx.coroutines.runBlocking
 
 class BuildInfoTest {
     @Test
-    fun `returns a valid build info`(): Unit = runBlocking {
-        val command = Document(mapOf("buildInfo" to 1))
-        val driver = Mockito.mock<MongoDbDriver>()
-        Mockito.`when`(driver.runCommand(command, BuildInfo::class)).thenReturn(BuildInfo("7.8.0", "1235abc"))
+    fun `returns a valid build info`(): Unit =
+        runBlocking {
+            val command = Document(mapOf("buildInfo" to 1))
+            val driver = Mockito.mock<MongoDbDriver>()
+            `when`(driver.connectionString()).thenReturn(ConnectionString("mongodb://localhost/"))
+            `when`(
+                driver.countAll("admin.atlascli".toNs(), Filters.eq("managedClusterType", "atlasCliLocalDevCluster")),
+            ).thenReturn(1L)
+            `when`(driver.runCommand("admin", command, BuildInfo::class)).thenReturn(
+                BuildInfo(
+                    "7.8.0",
+                    "1235abc",
+                    emptyList(),
+                    false,
+                    false,
+                    false,
+                    false,
+                    false,
+                    false,
+                    false,
+                    false,
+                    null,
+                    "mongodb://localhost",
```

Comment:
I think it's worth adding tests for things like multiple hostnames, hostnames that include ports, IPv6, etc. here :)